.jenkins/custom_pandoc_filter.py

```diff
@@ -0,0 +1,132 @@
+        if 'note' in classes:
+            color = '#54c7ec'
+            label = 'NOTE:'
+        elif 'tip' in classes:
+            color = '#6bcebb'
+            label = 'TIP:'
+        elif 'warning' in classes:
+            color = '#e94f3b'
+            label = 'WARNING:'
+        else:
+            return
+
+        note_content = []
+        for block in contents:
+            if 't' in block and block['t'] == 'Para':
+                for item in block['c']:
+                    if item['t'] == 'Str':
+                        note_content.append(Str(item['c']))
+                    elif item['t'] == 'Space':
+                        note_content.append(Space())
+                    elif item['t'] == 'Link':
+                        note_content.append(Link(*item['c']))
+                    elif item['t'] == 'Code':
+                        note_content.append(Code(*item['c']))
+            elif 't' in block and block['t'] == 'CodeBlock':
+                note_content.append(CodeBlock(*block['c']))
+
+        note_content_md = ''.join(to_markdown(item) for item in note_content)
+        html_content = markdown.markdown(note_content_md)
+
+        return [{'t': 'RawBlock', 'c': ['html', f'<div style="background-color: {color}; color: #fff; font-weight: 700; padding-left: 10px; padding-top: 5px; padding-bottom: 5px">{label}</div>']}, {'t': 'RawBlock', 'c': ['html', '<div style="background-color: #f3f4f7; padding-left: 10px; padding-top: 10px; padding-bottom: 10px; padding-right: 10px">']}, {'t': 'RawBlock', 'c': ['html', html_content]}, {'t': 'RawBlock', 'c': ['html', '</div>']}]
+    elif key == 'RawBlock':
+    # this is needed for the cells that have embedded video.
+    # We add a special tag to those: ``` {python, .jupyter-code-cell}
+    # The post-processing script then finds those and genrates separate
+    # code cells that can load video.
+        [format, content] = value
+        if format == 'html' and 'iframe' in content:
+            # Extract the video URL
+            video_url = content.split('src="')[1].split('"')[0]
+            # Create the Python code to display the video
+            python_code = f"""
+from IPython.display import display, HTML
+html_code = \"""
+{content}
+\"""
+display(HTML(html_code))
+"""
+
+            return {'t': 'CodeBlock', 'c': [['', ['python', 'jupyter-code-cell'], []], python_code]}
+
+
+def process_images(key, value, format, meta):
+    # Add https://pytorch.org/tutorials/ to images so that they
+    # load correctly in the notebook.
+    if key == 'Image':
+        [ident, classes, keyvals], caption, [src, title] = value
+        if not src.startswith('http'):
+            while src.startswith('../'):
+                src = src[3:]
+            if src.startswith('/_static'):
+                src = src[1:]
+            src = 'https://pytorch.org/tutorials/' + src
+        return {'t': 'Image', 'c': [[ident, classes, keyvals], caption, [src, title]]}
```

Comment:
Nit (explicitly return `None`, but return early) I might be wrong, in C++ it's considered a bad behavior to assume function will return None by default, but in Python it might be ok, though still a good thing to do implicitly
```suggestion
    if key != 'Image':
        return None
    [ident, classes, keyvals], caption, [src, title] = value
    if not src.startswith('http'):
        while src.startswith('../'):
            src = src[3:]
        if src.startswith('/_static'):
            src = src[1:]
        src = 'https://pytorch.org/tutorials/' + src
        
    return {'t': 'Image', 'c': [[ident, classes, keyvals], caption, [src, title]]}
```